doc/arkode/guide/source/Mathematics.rst

```diff
@@ -460,6 +460,100 @@ simplified form admits a more efficient and memory-friendly implementation
 than the more general form :eq:`ARKODE_IVP_simple_explicit`.
 
 
+.. _ARKODE.Mathematics.SPRKStep:
+
+SPRKStep -- Symplectic Partitioned Runge--Kutta methods
+=======================================================
+
+The SPRKStep time-stepping module in ARKODE is designed for IVPs of the form
+
+.. math::
+   \dot{p} = f_1(q,t) = \frac{\partial V(q,t)}{\partial q}, \quad 
+   \dot{q} = f_2(p) = \frac{\partial T(p)}{\partial p}, 
+   \qquad p(t_0) = p_0,\quad q(t_0) = q_0,
+   :label: ARKODE_IVP_Hamiltonian
+  
+where
+
+.. math::
+   H(p, q, t) = T(p) + V(q, t)
+
+is the system Hamiltonian. When :math:`dH/dt = 0`, i.e. when *H* is autonomous, then H is a conserved quantity. 
+Often this correponds to the conservation of energy (for example, in *n*-body problems).
+
+In solving the IVP :eq:`ARKODE_IVP_Hamiltonian`, we consider the problem in the form
+
+.. math::
+   \dot{y} = 
+      \begin{bmatrix}
+         f_1(q,t) \\
+         f_2(p)
+      \end{bmatrix} \qquad 
+      y(t_0) = 
+      \begin{bmatrix}
+         p_0\\
+         q_0
+      \end{bmatrix}
+
+SPRKStep utilizes Symplectic Partitioned Runge-Kutta (SPRK) methods represented by the pair of Butcher tableau,
+
+.. math::
+   \begin{array}{c|cccc}
+   c_1 & 0 & \cdots & 0 & 0 \\
+   c_2 & a_1 & 0 & \cdots & \vdots \\
+   \vdots & \vdots & \ddots & \ddots & \vdots \\
+   c_s & a_1 & \cdots & a_{s-1} & 0 \\
+   \hline
+   & a_1 & \cdots & a_{s-1} & a_s 
+   \end{array}
+   \qquad \qquad
+   \begin{array}{c|cccc}
+   C_1 & b_1 & \cdots & 0 & 0 \\
+   C_2 & b_1 & b_2 & \cdots & \vdots \\
+   \vdots & \vdots & \ddots & \ddots & \vdots \\
+   C_s & b_1 & b_2 & \cdots & b_{s-1} \\
+   \hline
+   & b_1 & b_2 & \cdots & b_{s-1}
+   \end{array}
+
+We use a compact storage of these coefficients in terms of two arrays, one for *a* and one for *b*.
+The time weights are computed on the fly. These methods approximately conserve a nearby Hamiltonian 
+for exponentially long times :cite:p:`HaWa:06`.  SPRKStep makes the assumption that the Hamiltonian is separable,
+in which case the schemes are explicit. SPRKStep provides methods with order of accuracy and conservation 
+of :math:`q = \{1,2,3,4,5,6,8,10\}`. The tables for these methods, and the default methods used, 
+are given in the section ???.
+
+In the default case, the algorithm for a single time-step is as follows
+
+#. Set :math:`P_0 = p_n, Q_1 = q_n`
+
+#. For :math:`i = 1,\ldots,s` do:
+
+   #. :math:`P_i = P_{i-1} + h_{n+1} a_i f_1(Q_i, t_n + C_i h_{n+1})`
+   #. :math:`Q_{i+1} = Q_i + h_{n+1} b_i f_2(P_i)`
+
+#. Set :math:`p_{n+1} = P_s, q_{n+1} = Q_{s+1}`
```

Comment:
There was inconsistency in my notation and I have changed it to use "c" and "\hat{c}" as well as "a" and "\hat{a}" instead of "a" and "b". The "\hat{c}" are used, but not the "c" right now, but its possible they will be in future extensions.